haystack/components/validators/json_schema.py

```diff
@@ -0,0 +1,172 @@
+import json
+from typing import List, Any, Dict, Optional
+
+from haystack import component
+from haystack.dataclasses import ChatMessage
+from haystack.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install jsonschema'") as jsonschema_import:
+    from jsonschema import validate, ValidationError
+
+
+@component
+class JsonSchemaValidator:
+    """
+    JsonSchemaValidator validates JSON content of ChatMessage against a specified JSON schema.
+
+    If JSON content of a message conforms to the provided schema, the message is passed along the "validated" output.
+    If the JSON content does not conform to the schema, the message is passed along the "validation_error" output.
+    In the latter case, the error message is constructed using the provided error_template or a default template.
+    These error ChatMessages can be used by LLMs in Haystack 2.x recovery loops.
+    """
+
+    # Default error description template
+    default_error_template = (
+        "The JSON content in the previous message does not conform to the provided schema.\n"
+        "Error details:\n- Message: {error_message}\n"
+        "- Error Path in JSON: {error_path}\n"
+        "- Schema Path: {error_schema_path}\n"
+        "Please match the following schema:\n"
+        "{json_schema}\n"
+        "and provide the corrected JSON content ONLY."
+    )
+
+    def __init__(self):
+        jsonschema_import.check()
+
+    @component.output_types(validated=List[ChatMessage], validation_error=List[ChatMessage])
+    def run(
+        self,
+        messages: List[ChatMessage],
+        json_schema: Dict[str, Any],
+        previous_messages: Optional[List[ChatMessage]] = None,
```

Comment:
we ask the user to pass `messages` and `previous_messages`,
but then we only process the last message in `messages`.

This suggests to me that the `messages` parameter might be enough: the last message is processed, while the others are automatically considered `previous_messages`. Am I missing something?